modules/javafx.base/src/main/java/com/sun/javafx/binding/ConditionalBinding.java

```diff
@@ -0,0 +1,67 @@
+package com.sun.javafx.binding;
+
+import java.util.Objects;
+
+import javafx.beans.value.ObservableValue;
+
+public class ConditionalBinding<T> extends LazyObjectBinding<T> {
+
+    private final ObservableValue<T> source;
+    private final ObservableValue<Boolean> nonNullCondition;
+
+    private Subscription subscription;
```

Comment:
I think it would be less good here, since I also need to know whether there currently is a subscription.  Comparing with the empty subscription feels not very nice, specifically this code:

    @Override
    protected T computeValue() {
        if (isObserved() && isActive()) {
            if (subscription.equals(Subscription.EMPTY)) {  // was: if(subscription == null) {
                subscription = Subscription.subscribeInvalidations(source, this::invalidate);
            }
        }
        else {
            unsubscribe();
        }

        return source.getValue();
    }

In flatMap this doesn't occur, and we don't care so much whether that specific subscription is present or not.  In flatMap the empty subscription is used because there always should be a subscription, but in case of a mapping to `null` there obviously can't be one.